### PR TITLE
[test] Fix chi examples and add them to CI

### PR DESCRIPTION
**Pull request by BillXu2000**
Related issue = #3630

Fix CHI example for current Taichi version, and add CHI example as a test to CI.
Not sure whether this works for platforms other than Ubuntu.

### CODE REVIEW

**Comment by BillXu2000** [comment]
CI is only set up for Ubuntu now.

**Comment by BillXu2000** [comment]
closing #3630

File: chi_examples/README.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ Follow the steps in `https://docs.taichi.graphics/lang/articles/contribution/dev
 
 Add option `-DTI_EXPORT_CORE=ON` to your `cmake` command (i.e. use `cmake .. DTI_EXPORT_CORE=ON`).
 
+Set environment variable `$TAICHI_REPO_DIR` to your Taichi repository path.
```
Comment (k-ye) [inline chi_examples/README.md:9]: Hmm, do we still need this?

File: tests/run_tests.py [hunk 2]
```diff
@@ -20,6 +20,29 @@ def _test_cpp():
         subprocess.check_call(f'./{cpp_test_filename}',
                               env={'TI_LIB_DIR': ti_lib_dir},
                               cwd=build_dir)
+        export_core_filename = 'libtaichi_export_core.so'
+        if os.path.exists(os.path.join(build_dir, export_core_filename)):
+            chi_dir = os.path.join(curr_dir, '../chi_examples')
+            subprocess.run('cmake . -B build',
```
Comment (k-ye) [inline tests/run_tests.py:26]: Could you use `[cmake, '.', '-B', 'build']` and make `shell=False`?

File: chi_examples/README.md (hunk 1 above)
Comment (ailzhang) [reply chi_examples/README.md:9]: no we should get rid of it :D

File: tests/run_tests.py (hunk 2 above)
Comment (BillXu2000) [reply tests/run_tests.py]: Sure, though I wonder the reason to switch it to `shell=False`.

**Comment by BillXu2000** [comment]
Problem has been solved in #3630, so I closed this PR.

**Pull request closed without merging**